index.js

```diff
@@ -23,42 +23,62 @@ module.exports = {
 		this._app = appOrAddon.app || appOrAddon;
 		this.options = new Config(this._app, this.ui);
 
-		this.importPolyfills();
+		this.importPolyfill();
 	},
 
-	importPolyfills() {
-		const { polyfillBundle, buildForProduction } = this.options;
+	importPolyfill() {
+		const { polyfillBundle } = this.options;
 
-		const webcomponentsjsPath = path.join(this._app.bowerDirectory, 'webcomponentsjs');
-		const customElementsEs5Adapter = path.join(webcomponentsjsPath, 'custom-elements-es5-adapter.js');
-		const webcomponentsjsPolyfill = path.join(webcomponentsjsPath, `webcomponents-${polyfillBundle}.js`);
+		if (polyfillBundle && polyfillBundle !== 'none') {
+			const webcomponentsjsPath = path.join(this._app.bowerDirectory, 'webcomponentsjs');
+			const webcomponentsjsPolyfill = path.join(webcomponentsjsPath, `webcomponents-${polyfillBundle}.js`);
 
-		// Import custom-elements-es5-adapter if ES6 Classes are transpiled to ES5
-		// for browsers that natively support Custom Elements
-		// https://github.com/webcomponents/webcomponentsjs#custom-elements-es5-adapterjs
-		if (buildForProduction.enabled && buildForProduction.build.js.compile) {
-			this._app.import(customElementsEs5Adapter, { options: 'prepend' });
+			this._app.import(webcomponentsjsPolyfill, { options: 'prepend' });
 		}
-
-		this._app.import(webcomponentsjsPolyfill, { options: 'prepend' });
 	},
 
 	// insert polyfills and bundled elements
 	contentFor(type, config) {
-		if (type !== 'head') {
-			return null;
-		}
+		if (type === 'head-footer') {
+			const headContents = [];
+			const { globalPolymerSettings, buildForProduction } = this.options;
 
-		const headContents = [];
-		const { globalPolymerSettings } = this.options;
+			if (globalPolymerSettings) {
+				headContents.push(`<script>window.Polymer = ${JSON.stringify(globalPolymerSettings)};</script>`);
+			}
 
-		if (globalPolymerSettings) {
-			headContents.push(`<script>window.Polymer = ${JSON.stringify(globalPolymerSettings)};</script>`);
+			if (buildForProduction.enabled && buildForProduction.build.js.compile) {
+				headContents.push(this.getCustomEs5Adapter());
+			}
+
+			return headContents.join('\n');
+		} else if (type === 'body-footer') {
+			return this.getBundleImport(config);
 		}
 
-		headContents.push(this.getBundleImport(config));
+		return null;
+	},
 
-		return headContents.join('\n');
+	getCustomEs5Adapter() {
+		const webcomponentsPolyfillsPath = path.join('assets', this._app.bowerDirectory, 'webcomponentsjs');
```

Comment:
The addon doesn't creates this directory inside the `dist/assets/` directory.
The polyfill raw code must be injected into the `index.html` file without using a file.

Reply:
polymer-project-bundler copies webcomponentsjs into assets
![Captura de pantalla 2019-03-28 a las 11 14 24](https://user-images.githubusercontent.com/92611/55149552-a8076f00-514a-11e9-977f-c0d5700756a8.png)

Reply:
I think polyfill raw injection is a more general solution. Anyhow, not a must right now

Reply:
The adapter should be included conditionally depending on browser capabilities (support for `customElements`) and that cannot be done at build time, so a script should be injected / imported and that script should exist. The script cannot be inlined. This is how polymer build works: https://polymer-library.polymer-project.org/3.0/docs/apps/build-for-production